Sustainsys.Saml2/WebSSO/AcsCommand.cs

```diff
@@ -45,16 +46,22 @@ public CommandResult Run(HttpRequestData request, IOptions options)
                 try
                 {
                     unbindResult = binding.Unbind(request, options);
+
                     options.Notifications.MessageUnbound(unbindResult);
 
                     var samlResponse = new Saml2Response(unbindResult.Data, request.StoredRequestState?.MessageId, options);
+                    
+                    var idpContext = GetIdpContext(unbindResult.Data, options);
+
+                    var result = ProcessResponse(options, samlResponse, request.StoredRequestState, idpContext, unbindResult.RelayState);
 
-                    var result = ProcessResponse(options, samlResponse, request.StoredRequestState);
-                    if(unbindResult.RelayState != null)
+                    if(unbindResult.RelayState != null && (idpContext != null ? !idpContext.RelayStateUsedAsReturnUrl : false))
```

Comment:
This expression doesn't look right. What if SP-initiated sign in is used with an Idp that has RelayStateUsedAsReturnUrl=true? An Idp could support both SP-initiated sign in (where RelayState should be just relayed) and Idp-initiated where the RelayState should be treated as a ReturnUrl.

Reply:
By using Saml2Notifications.GetIdentityProvider the idpContext is not going to be null and so the expression can be simplified. Please see the latest commit.